src/validation/rooms.ts

```diff
@@ -0,0 +1,38 @@
+import { RoomInterface } from '../entity/Room';
+import {
+  ERROR_ROOM_CREATE_EMPTY_DATA,
+  ERROR_ROOM_CREATE_EMPTY_OPTIONS,
+} from '../routes/strings';
+
+export enum ValidationRoomStatus {
+  OK,
+  INVALID,
+}
+
+export interface RoomValidationResult {
+  error?: string;
+  message?: string;
+  status: ValidationRoomStatus;
+}
+
+export function validateRoom(newRoom: RoomInterface): RoomValidationResult {
```

Comment:
I like this validation approach better, what is unclear to me now is, what kind of validations are we doing here? validations for incoming rooms, before saving, updating, etc.

Just asking because in the [controller we are also "validating" the model](https://github.com/apipath/ppoker-rest-js/pull/8/files#diff-b0683cfec0a2766af16effa581369f29L82), should we have different validators? maybe u just forgot to add that? what's the goal of this abstraction?

Reply:
> I like this validation approach better, what is unclear to me now is, what kind of validations are we doing here? validations for incoming rooms, before saving, updating, etc.

I didn't think about other possible validations. This was only a simple approach to separate "validation" logic that's tied to the model from the route handling (in this way the validation is easier to test and read). We can expand to more use cases if you think it's fine. 

> Just asking because in the controller we are also "validating" the model, should we have different validators? maybe u just forgot to add that? what's the goal of this abstraction?

I'm not seeing from the link you provided that we are validating twice.  
Here we imperatively validate and emit a result, then in the route handler we just consume that result but the function itself doesn't know "how" to perform any validation.

The goal of this abstraction to isolate complexity.